src/base/bittorrent/session.cpp

```diff
@@ -1464,9 +1502,17 @@ void Session::configure(libtorrent::session_settings &sessionSettings)
     sessionSettings.enable_outgoing_utp = isUTPEnabled();
     // uTP rate limiting
     sessionSettings.rate_limit_utp = isUTPRateLimited();
-    sessionSettings.mixed_mode_algorithm = isUTPRateLimited()
-                                           ? libt::session_settings::prefer_tcp
-                                           : libt::session_settings::peer_proportional;
+    switch(utpMixedMode()) {
```

Comment:
You should check/normalize values when getting it from untrusted sources (e.g. from config file). Property getter always should return valid value.